### PR TITLE
Heatmap w/o cities w/ small R2,reversed R2 clbar

### PR DESCRIPTION
**Pull request by karukes**
#88

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -151,12 +155,20 @@ def generate_predictions(methods: list[str], datasets: list[str]) -> Generator[R
...
             if len(predictions_per_dataset) > 0:
                 comparison_plot_multi(dataset.values.loc[training_index, :], predictions_per_dataset)
 
-            results_store.append(reports_per_dataset)
-            yield reports_per_dataset
-        
+            #appends only the cities where R2 > 10
+            if (np.array(metrics_stat)>-10).all():
+                results_store.append(reports_per_dataset)
+                yield reports_per_dataset
```
Comment (swhustla) [inline main.py:170]: `yield  reports_per_dataset` doesn't need to be indented- it's only the results_store we want to block.

**Comment by swhustla** [comment]
Hi @karukes I've just mentioned a small change to the `main.py` file

File: main.py (hunk 1 above)
Comment (karukes) [reply main.py]: Right, it got accidentally in the middle. I have just changed it.

File: main.py [hunk 2]
```diff
@@ -150,13 +154,22 @@ def generate_predictions(methods: list[str], datasets: list[str]) -> Generator[R
 
                 predictions_per_dataset.append(report.prediction)
                 reports_per_dataset.append(report)
+                yield reports_per_dataset
```
Comment (swhustla) [inline main.py:157]: Ah, now I think it's indented too far. `yield` here will break out of the loop too early.
It needs to be the very last thing executed in the `for dataset_name in datasets:` loop.

**Review by swhustla — CHANGES_REQUESTED** [review]
The yield needs to move again, I'm afraid. Hope it makes sense!

File: main.py (hunk 2 above)
Comment (swhustla) [reply main.py:157]: Ideally:
```
if (np.array(metrics_stat)>-10).all():
                results_store.append(reports_per_dataset)
yield reports_per_dataset
```

Comment (swhustla) [reply main.py:157]: So at the same indentation as the if statement stub.

Comment (karukes) [reply main.py]: Done, should be ok now.

**Review by swhustla — APPROVED** [review]
Looks good!

**Pull request merged**